server/src/main/java/com/distkv/server/metaserver/client/DmetaClient.java

```diff
@@ -1,87 +1,61 @@
 package com.distkv.server.metaserver.client;
 
 import com.alipay.remoting.exception.RemotingException;
+import com.alipay.sofa.jraft.JRaftUtils;
 import com.alipay.sofa.jraft.RouteTable;
 import com.alipay.sofa.jraft.conf.Configuration;
 import com.alipay.sofa.jraft.entity.PeerId;
 import com.alipay.sofa.jraft.option.CliOptions;
 import com.alipay.sofa.jraft.rpc.impl.cli.BoltCliClientService;
-import com.distkv.server.metaserver.server.bean.GetRequest;
-import com.distkv.server.metaserver.server.bean.GetResponse;
-import com.distkv.server.metaserver.server.bean.PutRequest;
-import com.distkv.server.metaserver.server.bean.PutResponse;
+import com.distkv.common.NodeInfo;
+import com.distkv.server.metaserver.server.bean.HeartbeatRequest;
+import com.distkv.server.metaserver.server.bean.HeartbeatResponse;
 
 import java.util.concurrent.TimeoutException;
 
 public class DmetaClient {
 
   private BoltCliClientService cliClientService = null;
 
-  public static final String groupId = "KV";
+  // TODO(qingw): Refine this.
+  public static final String GROUP_ID = "KV";
 
-  public static final String confStr = "127.0.0.1:8081,127.0.0.1:8082,127.0.0.1:8083";
+  public DmetaClient(String confStr) throws TimeoutException, InterruptedException {
 
-  public DmetaClient() {
+    final Configuration conf = JRaftUtils.getConfiguration(confStr);
 
-    final Configuration conf = new Configuration();
-    if (!conf.parse(confStr)) {
-      throw new IllegalArgumentException("Fail to parse conf:" + confStr);
-    }
-
-    RouteTable.getInstance().updateConfiguration(groupId, conf);
+    RouteTable.getInstance().updateConfiguration(GROUP_ID, conf);
 
     //init RPC client and update Routing table
     cliClientService = new BoltCliClientService();
     cliClientService.init(new CliOptions());
-  }
-
-  public void put(String key, String value) {
-    try {
-      if (!RouteTable.getInstance().refreshLeader(cliClientService, groupId, 1000).isOk()) {
-        throw new IllegalStateException("Refresh leader failed");
-      }
-      //get leader term
-      final PeerId leader = RouteTable.getInstance().selectLeader(groupId);
-
-      PutRequest request = new PutRequest();
-      request.setKey(key);
-      request.setValue(value);
-      PutResponse response = (PutResponse) cliClientService.getRpcClient()
-          .invokeSync(leader.getEndpoint().toString(), request, 3000);
-      if (!response.isSuccess()) {
-        throw new RuntimeException("put error");
-      }
-    } catch (InterruptedException e) {
-      e.printStackTrace();
-    } catch (TimeoutException e) {
-      e.printStackTrace();
-    } catch (RemotingException e) {
-      e.printStackTrace();
+    //refresh leader term.
+    if (!RouteTable.getInstance().refreshLeader(cliClientService, GROUP_ID, 1000).isOk()) {
+      throw new IllegalStateException("Refresh leader failed");
     }
   }
 
-  public String get(String key) {
+  public HeartbeatResponse heartbeat(NodeInfo nodeInfo, int timeout) {
```

Comment:
After my thinking again, it's actually unreasonable to heartbeat the nodeInfo. How about just using node id instead of it?

Reply:
Timeout should not be a parameter either.